lib/containers/utils.pm

```diff
@@ -40,27 +40,59 @@ sub test_seccomp {
     }
 }
 
-sub check_docker_firewall {
-    my $container_name = 'sut_container';
-    my $docker_version = get_docker_version();
-    systemctl('is-active firewalld');
-    my $running = script_output qq(docker ps -q | wc -l);
-    validate_script_output('ip a s docker0', sub { /state DOWN/ }) if $running == 0;
-    # Docker zone is created for docker version >= 20.10 (Tumbleweed), but it
-    # is backported to docker 19 for SLE15-SP3 and for Leap 15.3
-    if (check_runtime_version($docker_version, ">=20.10") || is_sle('>=15-sp3') || is_leap(">=15.3")) {
-        assert_script_run "firewall-cmd --list-all --zone=docker";
-        validate_script_output "firewall-cmd --list-interfaces --zone=docker", sub { /docker0/ };
+sub check_containers_firewall {
+    my ($runtime) = @_;
+    record_info "firewall", "Checking that firewall is enabled, properly configured and containers can reach the Internet";
+    if ($runtime eq 'docker') {
+        my $container_name = 'sut_container';
+        my $docker_version = get_docker_version();
+        systemctl('is-active firewalld');
+        my $running = script_output qq(docker ps -q | wc -l);
+        validate_script_output('ip a s docker0', sub { /state DOWN/ }) if $running == 0;
+        # Docker zone is created for docker version >= 20.10 (Tumbleweed), but it
+        # is backported to docker 19 for SLE15-SP3 and for Leap 15.3
+        if (check_runtime_version($docker_version, ">=20.10") || is_sle('>=15-sp3') || is_leap(">=15.3")) {
+            assert_script_run "firewall-cmd --list-all --zone=docker";
+            validate_script_output "firewall-cmd --list-interfaces --zone=docker", sub { /docker0/ };
```

Comment:
I would also check here, if `docker0` is in the active firewall zones, e.g.

    validate_script_output "firewall-cmd --get-active-zones", sub { /docker.*interfaces: docker0/ };

or something similar. WDYT?